### DISCUSSION TITLE
[OpenAPI] How to remove and customize x-codeSamples?

### DISCUSSION THREAD

**Discussion by devnjw** [fuma-nama/fumadocs#721] [Q&A] [answered]
Hi, thank you for the awesome open source project.

I want to customize `x-codeSamples`. I only want `cURL`, `Python`, and `Langchain` to be included.
I know I can add languages with `x-codeSamples` openapi spec, but I'm not sure how to remove the default samples.

![image](https://github.com/user-attachments/assets/83d9d95e-6fe7-4edb-bb35-6bcd42a8b63d)

**Comment by devnjw** [comment]
Also, is there a way to hide the code sample and response sample cards on the right side?

**Comment by devnjw** [comment]
@fuma-nama Hello. Do you have any plan for custom OpenAPI code samples?

**Reply by fuma-nama** [reply, marked as answer]
already supported, please refer to the docs

**Reply by devnjw** [reply]
Thank you for reply. I'll find the docs :)

**Reply by devnjw** [reply]
@fuma-nama 
Is it possible to remove the `APIExample` from the `APIPage`?

**Reply by fuma-nama** [reply]
Remove code samples of a specific label by adding a code sample with the same label, and set source to false/undefined.
To remove all samples, you have to disable them one by one

**Reply by devnjw** [reply]
@fuma-nama Thanks! I found your [example code](https://github.com/fuma-nama/fumadocs/blob/8fe83e10bd3be8a7fe9ad50501b8007493e072fc/apps/docs/museum.yaml#L71-L79).

It would be awesome if I could remove response tab too.
![image](https://github.com/user-attachments/assets/a7fe7164-6549-465e-b15e-b922933e33cd)

**Reply by fuma-nama** [reply]
Why? This doesn't make sense to me

**Reply by devnjw** [reply]
Sometimes I only want to use API Request Test Block.
![image](https://github.com/user-attachments/assets/3df8f1da-3325-4383-99d5-cba08fbf669c)

**Reply by devnjw** [reply]
I want to put toc instead, but toc with response example looks bad.

**Reply by fuma-nama** [reply]
in this case you should just override the renderer of `API` to change its layout (e.g. using `flex-col`)
